x-pack/plugins/cloud_security_posture/server/routes/csp_rule_template/update_csp_rules_states.ts

```diff
@@ -0,0 +1,170 @@
+    INTERNAL_CSP_SETTINGS_SAVED_OBJECT_ID,
+    { rules_states: newRulesStates }
+  );
+};
+
+export const setRulesStates = (
+  rulesStates: CspRulesStates,
+  ruleIds: string[],
+  state: boolean
+): CspRulesStates => {
+  ruleIds.forEach((ruleId) => {
+    if (rulesStates[ruleId]) {
+      // Rule exists, set entry
+      rulesStates[ruleId] = { muted: state };
+    } else {
+      // Rule does not exist, create an entry
+      rulesStates[ruleId] = { muted: state };
+    }
+  });
+  return rulesStates;
+};
+
+export const createCspSettingObject = async (soClient: SavedObjectsClientContract) => {
+  return soClient.create<CspSettings>(
+    INTERNAL_CSP_SETTINGS_SAVED_OBJECT_TYPE,
+    {
+      rules_states: {},
+    },
+    { id: INTERNAL_CSP_SETTINGS_SAVED_OBJECT_ID }
+  );
+};
+
+export const createCspSettingObjectSafe = async (
+  soClient: SavedObjectsClientContract,
+  logger: Logger
+) => {
+  const cspSettings = await getCspSettings(soClient, logger);
+
+  if (!cspSettings) return (await createCspSettingObject(soClient)).attributes;
+};
+
+export const getCspSettings = async (
+  soClient: SavedObjectsClientContract,
+  logger: Logger
+): Promise<CspSettings | undefined> => {
+  try {
+    const cspSettings = await soClient.get<CspSettings>(
+      INTERNAL_CSP_SETTINGS_SAVED_OBJECT_TYPE,
+      INTERNAL_CSP_SETTINGS_SAVED_OBJECT_ID
+    );
+    return cspSettings.attributes;
+  } catch (err) {
+    const error = transformError(err);
+    if (error.statusCode === 404) {
+      return undefined;
+    } else {
+      logger.error(`An error occurred while trying to fetch csp settings: ${error}`);
+    }
+  }
+};
+
+export const getCspSettingsSafe = async (
+  soClient: SavedObjectsClientContract,
+  logger: Logger
+): Promise<CspSettings> => {
+  const cspSettings = await getCspSettings(soClient, logger);
+
+  if (!cspSettings) return (await createCspSettingObject(soClient)).attributes;
+
+  return cspSettings;
+};
+
+const buildRuleKey = (benchmarkId: string, benchmarkVersion: string, ruleNumber: string) => {
+  return `${benchmarkId};${benchmarkVersion};${ruleNumber}`;
+};
+
+export const defineUpdateCspRuleStateRoute = (router: CspRouter) =>
+  router.versioned
+    .post({
+      access: 'internal',
+      path: UPDATE_CSP_RULE_STATE_ROUTE_PATH,
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: {
+            body: cspRuleBulkActionRequest,
+          },
+        },
+      },
+      async (context, request, response) => {
+        if (!(await context.fleet).authz.fleet.all) {
+          return response.forbidden();
+        }
+        const cspContext = await context.csp;
+
+        try {
+          const requestBody: CspRuleBulkActionRequest = request.body;
+
+          const cspSettings = await getCspSettings(cspContext.soClient, cspContext.logger);
+
+          if (!cspSettings) {
+            throw cspContext.logger.error(`Failed to read csp settings`);
+          }
```

Comment:
I recommend not to fail the action in this case - it will leave the user blocked with no way to recover

So it is always better to overcome this situation, which I believe we can